dash/dash.py

```diff
@@ -86,106 +86,223 @@ class _NoUpdate(object):
 # pylint: disable=too-many-instance-attributes
 # pylint: disable=too-many-arguments, too-many-locals
 class Dash(object):
+    """
```

Comment:
😸  nice docstring.

 tangential question, do we have an agreed reference to follow?  I feel like it's not very helpful to use super-long-detailed-sphinx-style; chris is also building sth in the house.  

but from our users' perspective, data scientists might feel this more familiar https://sphinxcontrib-napoleon.readthedocs.io/en/latest/

Reply:
I'd be up for changing to a cleaner style, I was just copying the style we already had for `run_server` and `enable_dev_tools`.

Feels to me like we should have a decent level of detail here, and then use this to generate the corresponding page in dash-docs. Let's keep this format until we decide how we want to do that, so we only need to alter it once. @chriddyp I'm not aware if you're working on something else to fill this gap.